data/lua/core/interface.lua

```diff
@@ -36,6 +37,50 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 		moving_unit:to_map(to_x, to_y)
 		wesnoth.wml_actions.redraw{}
 	end
+	---Fakes the move of a unit satisfying the given filter to position x, y.
+    ---Usable only during WML actions.
+	---Todo : maybe implement with C++.
+	---@param to_x integer
+    ---@param to_y integer
+    ---@param u unit
+	---@param queued boolean
+    function wesnoth.interface.move_unit_fake_queue(u, to_x, to_y, queued)
+        ---Fakes the move of a given unit, can be queued.
+		if to_x and to_y and u then
+			table.insert(wesnoth.interface.queued_movements, { to_x = to_x, to_y = to_y, u = u })
+		end
+		if queued then
+			return
+		end
+
+        for _index, move in ipairs(wesnoth.interface.queued_movements) do
+            local moving_unit = move.u
+            local from_x, from_y = moving_unit.x, moving_unit.y
+
+            wesnoth.interface.scroll_to_hex(from_x, from_y)
```

Comment:
I'm not sure if it's necessary to actually scroll to the hex… I guess it can stay for now…

Reply:
I'm not sure… I think scrolling maybe shouldn't be directly linked to an animation, but in this case it certainly might make sense that you'd want to scroll the screen so you can see the full extent of the movement. But it also might make sense that your goal is to make the unit move offscreen and disappear. So probably _not_ scrolling is the more versatile option.